src/platform/Linux/bluez/BluezAdvertisement.cpp

```diff
@@ -224,39 +221,49 @@ void BluezAdvertisement::Shutdown()
 
 void BluezAdvertisement::StartDone(GObject * aObject, GAsyncResult * aResult)
 {
-    auto * advMgr = reinterpret_cast<BluezLEAdvertisingManager1 *>(aObject);
     GAutoPtr<GError> error;
-    gboolean success = FALSE;
+    gboolean success = bluez_leadvertising_manager1_call_register_advertisement_finish(
+        reinterpret_cast<BluezLEAdvertisingManager1 *>(aObject), aResult, &error.GetReceiver());
 
-    success = bluez_leadvertising_manager1_call_register_advertisement_finish(advMgr, aResult, &error.GetReceiver());
-    VerifyOrExit(success == TRUE, ChipLogError(DeviceLayer, "FAIL: RegisterAdvertisement : %s", error->message));
+    VerifyOrReturn(success == TRUE, {
```

Comment:
This macro use seems to make things harder to read. How about having a `if (!success) { ... return;} instead?

Reply:
> on failure is error guaranteed non-null?

Yes, glib sets error and returns NULL in the `g_dbus_proxy_call_finish()` call which is used by all `bluez_*_finish()` wrappers.

https://developer-old.gnome.org/gio/stable/GDBusProxy.html#g-dbus-proxy-call-finish